File: frontend/types/intake.ts
export interface SessionStartRequest { encounter_id: string; language: string; schema_id: string; }
export interface SessionStartResponse { session_id: string; encounter_id: string; schema_id: string; language: string; first_question: string | null; first_question_field_name: string | null; status: string; }
export interface IntakeTurnRequest { session_id: string; encounter_id: string; touch_answer?: string | null; answering_field_name?: string | null; }
export interface ExtractedEntitySummary { field_name: string; value: string; confidence: number; low_confidence_flag: boolean; source_type: string; source_location: string | null; }
export interface IntakeTurnResponse { session_id: string; next_question: string | null; next_question_field_name: string | null; pathway_complete: boolean; entities_extracted: ExtractedEntitySummary[]; turn_number: number; raw_transcript: string | null; detected_language: string | null; }
export type DocumentType = "prescription" | "lab_report" | "discharge_summary";
export interface DocumentUploadResponse {
  document_id: string;
  encounter_id: string;
  document_type: string;
  original_filename?: string | null;
  processing_status?: string | null;
  processing_error?: string | null;
  file_size?: number | null;
  entities_extracted: ExtractedEntitySummary[];
  entity_count: number;
}
export interface IntakeSubmitRequest { session_id: string; encounter_id: string; }
export interface IntakeSubmitResponse { encounter_id: string; session_id: string; status: string; total_entities: number; timeline_events: number; summary_preview: string; }

File: backend/schemas/intake.py
"""
backend/schemas/intake.py

Pydantic request/response schemas for all Phase 2 patient intake endpoints:
  POST /intake/session/start
  POST /intake/turn
  POST /intake/document/upload   (uses FastAPI Form/File — not a JSON body)
  POST /intake/submit
"""

from __future__ import annotations
from pydantic import BaseModel, Field
from typing import Optional, List


# ---------------------------------------------------------------------------
# POST /intake/session/start
# ---------------------------------------------------------------------------
class SessionStartRequest(BaseModel):
    encounter_id: str
    language: str = Field(default="en", description="BCP-47 language code, e.g. 'en', 'hi'")
    schema_id: str = Field(
        default="allopathic_chest_pain_v1",
        description="Clinical schema to use for adaptive questioning",
    )


class SessionStartResponse(BaseModel):
    session_id: str
    encounter_id: str
    schema_id: str
    language: str
    # The first question to show the patient immediately
    first_question: Optional[str]
    first_question_field_name: Optional[str]
    status: str = "in_progress"


# ---------------------------------------------------------------------------
# POST /intake/turn
# ---------------------------------------------------------------------------
class IntakeTurnRequest(BaseModel):
    session_id: str
    encounter_id: str
    touch_answer: Optional[str] = Field(
        default=None,
        description="Text input from touch/keyboard (or pre-transcribed ASR text)",
    )
    answering_field_name: Optional[str] = Field(
        default=None,
        description=(
            "Which schema field this answer responds to. "
            "Must match the prior response's next_question_field_name. "
            "On the very first turn, use the first_question_field_name from session/start."
        ),
    )


class ExtractedEntitySummary(BaseModel):
    """Lightweight entity summary returned to the patient UI (no DB ids)."""
    field_name: str
    value: str
    confidence: float
    low_confidence_flag: bool
    source_type: str
    source_location: Optional[str]


class IntakeTurnResponse(BaseModel):
    session_id: str
    next_question: Optional[str]
    next_question_field_name: Optional[str]
    pathway_complete: bool
    entities_extracted: List[ExtractedEntitySummary]
    turn_number: int
    raw_transcript: Optional[str] = None
    detected_language: Optional[str] = None


# ---------------------------------------------------------------------------
# POST /intake/document/upload  (multipart — schema shown for docs only)
# ---------------------------------------------------------------------------
class DocumentUploadResponse(BaseModel):
    document_id: str
    encounter_id: str
    document_type: str
    original_filename: Optional[str] = None
    processing_status: str = "processed"
    processing_error: Optional[str] = None
    file_size: Optional[int] = None
    entities_extracted: List[ExtractedEntitySummary]
    entity_count: int


# ---------------------------------------------------------------------------
# POST /intake/submit
# ---------------------------------------------------------------------------
class IntakeSubmitRequest(BaseModel):
    session_id: str
    encounter_id: str


class IntakeSubmitResponse(BaseModel):
    encounter_id: str
    session_id: str
    status: str                       # "ready_for_review"
    total_entities: int
    timeline_events: int
    summary_preview: str              # first 200 chars of the generated summary
